### PR TITLE
Scala 3 Migration

### PR DESCRIPTION
**Pull request by LaurelineSwissBorg**

### CODE REVIEW

File: sjsonnet/src-jvm-native/sjsonnet/CachedResolvedFile.scala [hunk 1]
```diff
@@ -61,12 +61,12 @@ class CachedResolvedFile(val resolvedImportPath: OsPath, memoryLimitBytes: Long,
     }
   }
 
-  override lazy val contentHash: String = {
```
Comment (lomigmegard) [inline sjsonnet/src-jvm-native/sjsonnet/CachedResolvedFile.scala]: we loose the caching here. Any way to restore the lazy val logic?

File: sjsonnet/src/sjsonnet/Evaluator.scala [hunk 2]
```diff
@@ -502,14 +502,20 @@ class Evaluator(resolver: CachedResolver,
   def visitMemberList(objPos: Position, e: ObjBody.MemberList, sup: Val.Obj)(implicit scope: ValScope): Val.Obj = {
     val asserts = e.asserts
     val fields = e.fields
-    var cachedSimpleScope: ValScope = null.asInstanceOf[ValScope]
+    var cachedSimpleScope: ValScope | Null = null
```
Comment (tiagomota) [inline sjsonnet/src/sjsonnet/Evaluator.scala:505]: question: Will this work in the same wya as before? 🤔

File: sjsonnet/src/sjsonnet/Importer.scala [hunk 3]
```diff
@@ -158,7 +158,7 @@ case class StaticResolvedFile(content: String) extends ResolvedFile {
   def readString(): String = content
 
   // We just cheat, the content hash can be the content itself for static imports
-  lazy val contentHash: String = content
+  def contentHash(): String = content
```
Comment (tiagomota) [inline sjsonnet/src/sjsonnet/Importer.scala]: question: Here is another swap from `lazy val` to `def` as @lomigmegard pointed. What was the reasoning to change it?

File: sjsonnet/src/sjsonnet/Evaluator.scala (hunk 2 above)
Comment (LaurelineSwissBorg) [reply sjsonnet/src/sjsonnet/Evaluator.scala:505]: Yes. It was failing all the tests due to some changes in how Scala 3 handles AnyVal. I had to find a new way to express the same behavior and this is the result.

File: sjsonnet/src-jvm-native/sjsonnet/CachedResolvedFile.scala (hunk 1 above)
Comment (LaurelineSwissBorg) [reply sjsonnet/src-jvm-native/sjsonnet/CachedResolvedFile.scala]: I had to modify the parent trait to use a parens-less function.

**Review by tiagomota — APPROVED** [review]

**Pull request merged**